cmd/gateway/setup.go

```diff
@@ -120,3 +123,22 @@ func MustValidateArguments(flagset *flag.FlagSet, validators ...ValidatorContext
 		os.Exit(1)
 	}
 }
+
+func getBuildInfo() (commitHash string, commitTime string, dirtyBuild bool) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "unknown", "unknown", true
```

Comment:
why do we set `dirtyBuild` to true in this case rather than "unknown"?

Reply:
it just seems inconsistent to me. for first two, we report 'unknown', but for dirty we report false, so we sort of making an assumption.

Reply:
I guess that if we just use the value to print it in the logs it makes sense to have it as a string?

Reply:
+1 to string

No need for a bool if all we are doing is printing the value to the logs.

Reply:
converted to a string